src/main/java/ch/njol/skript/sections/EffSecSendResourcePack.java

```diff
@@ -0,0 +1,214 @@
+	@Nullable
+	private Trigger trigger;
+	@Nullable
+	private Expression<String> oldHash;
+	private int pattern;
+
+	@SuppressWarnings({"unchecked", "null"})
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult, @Nullable SectionNode sectionNode, List<TriggerItem> triggerItems) {
+		this.url = (Expression<String>) exprs[0];
+		this.pattern = matchedPattern;
+		if (matchedPattern == 2) {
+			this.oldHash = (Expression<String>) exprs[1];
+			this.recipients = (Expression<Player>) exprs[2];
+		} else {
+			this.recipients = (Expression<Player>) exprs[1];
+		}
+
+		if (sectionNode != null) {
+			AtomicBoolean delayed = new AtomicBoolean(false);
+			Runnable afterLoading = () -> delayed.set(!getParser().getHasDelayBefore().isFalse());
+			trigger = loadCode(sectionNode, "resource", afterLoading, ResourcePackEvent.class);
+			if (delayed.get()) {
+				Skript.error("Delays can't be used within a Resource Pack Effect Section");
+				return false;
+			}
+		}
+
+		return true;
+	}
+
+
+	@Override
+	protected @Nullable TriggerItem walk(Event event) {
+		debug(event, true);
+		assert url != null;
+		String address = url.getSingle(event);
+		UUID uuid = null;
+		String hash =  this.oldHash == null ? "defaultHash" : this.oldHash.getSingle(event);
+		String prompt = "The server has requested you to download a resource pack";
+		boolean force = false;
+		if (trigger != null) {
+			ResourcePackEvent resourcePackEvent = new ResourcePackEvent();
+			Variables.setLocalVariables(resourcePackEvent, Variables.copyLocalVariables(event));
+			TriggerItem.walk(trigger, resourcePackEvent);
+			Variables.setLocalVariables(event, Variables.copyLocalVariables(resourcePackEvent));
+			Variables.removeLocals(resourcePackEvent);
+			String checkUUID = resourcePackEvent.getId();
+			String checkHash = resourcePackEvent.getHash();
+			String checkPrompt = resourcePackEvent.getPrompt();
+			Boolean checkForce = resourcePackEvent.getForce();
+
+			if (checkUUID != null) {
+				uuid = UUID.fromString(Utils.convertUUID(checkUUID));
+			}
+			if (checkHash != null) {
+				hash = checkHash;
+			}
+			if (checkPrompt != null) {
+				prompt = checkPrompt;
+			}
+			if (checkForce) {
+				force = checkForce;
+			}
+		}
+		for (Player player : recipients.getArray(event)) {
+			if (uuid != null) {
+				player.setResourcePack(uuid, address, StringUtils.hexStringToByteArray(hash), prompt, force);
+			} else {
+				player.setResourcePack(address, StringUtils.hexStringToByteArray(hash), prompt, force);
+			}
+		}
+		return super.walk(event, false);
+	}
+
+
+	@Override
+	public String toString(@Nullable Event event, boolean debug) {
+		String result = "send the resource pack from the URL " + url.toString(event, debug);
+		if (this.oldHash != null) {
+			result += " with hash " + this.oldHash.toString(event, debug);
+		}
+		if (trigger != null) {
+			ResourcePackEvent resourcePackEvent = (ResourcePackEvent) event;
```

Comment:
The even will never be a resourcepackevent unless you shoved this section within another resource pack section, in which case it will get the wrong values. This simply should not toString its section, just the effect part.